lib/internal/url.js

```diff
@@ -7,6 +7,8 @@ const context = Symbol('context');
 const cannotBeBase = Symbol('cannot-be-base');
 const special = Symbol('special');
 const searchParams = Symbol('query');
+const urlInstance = Symbol('url-instance');
+const searchParamsInstance = Symbol('searchParams-instance');
```

Comment:
Why not use the existing `context` and `searchParams` symbols rather than introducing new ones? All `URL` objects should have the `context` symbol set. Likewise all `URLSearchParams` instances should have the `searchParams` symbol set (I believe)